### PR TITLE
:wrench: Page switcher now shows all active locales

### LINKED ISSUE #372: Behavior for pages not available in all languages
For a given page, the language select dropdown only offers the languages that this article is available in.

If a user lands on a page in a language they don't speak, they might want to change the language so they can navigate the page.
Should we consider this case and instead of not offering all languages, rather redirect to the home in the language of choice, or a page indicating that this article doesn't exist in the language of choice?

### PR DESCRIPTION
**Pull request by ehmadzubair**
Closes #372

### CODE REVIEW

File: home/templatetags/home_tags.py [hunk 1]
```diff
@@ -1,38 +1,31 @@
...
-
-    default_locales = []
+    switcher_locales = list()
     locales = Locale.objects.select_related('locale_detail').all()
     for locale in locales:
+        should_append = False
         try:
             if locale.locale_detail.is_active:
-                default_locales.append(locale)
+                should_append = True
```
Comment (sheralim012) [inline home/templatetags/home_tags.py:20]: should_append = locale.locale_detail.is_active

**Review by sheralim012 — CHANGES_REQUESTED** [review]

**Comment by geoo89** [comment]
In the MZ migration, it appears that only the current language is displayed in the picker.
On the English (blank) homepage, I cannot select Portuguese, and vice versa. On any Portuguese article, I cannot select English.

Note that if an article is not available in a certain language, the user should be redirected to a page indicating something like "Sorry, '[page title]' is not available in [language]. If you want to keep browsing in [language] you can click here to return the homepage."

**Comment by ehmadzubair** [comment]
In #372 

> Remark: This is a temporary solution unless we have dedicated page (and translations of it) to indicate that the page is not available in the language of choice.

I was of the impression that we are going for the temporary solution?

**Comment by geoo89** [comment]
We discussed this with Chris and redirecting to the homepage is confusing.

We will add the strings needed for the dedicated page to the translation manager for the country offices to translate.

**Comment by ehmadzubair** [comment]
@geoo89 I've added the intermediate page for unavailable translations. 

There is however an edge case here. If we change the language of the intermediate page, technically we are redirected to the intermediate page in another language. While it is technically correct, the user experience is broken  because the original page can be available in the selected language.

What do you think we should do here?

**Comment by geoo89** [comment]
One minor issue: If a page is live in one language A, but draft in another language B, then going to the article in language A and then changing to language B gives a 404 rather than the "language not available page".

> There is however an edge case here. If we change the language of the intermediate page, technically we are redirected to the intermediate page in another language. While it is technically correct, the user experience is broken because the original page can be available in the selected language.
> 
> What do you think we should do here?

I think it'd probably be best to go back to the article that the user was trying to view in the unavailable language.
@cbunicef to confirm.

Currently it seems to crash in this case?

```
Environment:


Request Method: GET
Request URL: http://localhost:8000/en/translation-not-found/?prev=/ny/translation-not-found/&page=

Django Version: 3.1.13
Python Version: 3.8.0
Installed Applications:
['home',
 'search',
 'iogt_users',
 'comments',
 'iogt_content_migration',
 'questionnaires',
 'messaging',
 'django.contrib.humanize',
 'wagtail_localize',
 'wagtail_localize.locales',
 'wagtail.contrib.forms',
 'wagtail.contrib.redirects',
 'wagtail.embeds',
 'wagtail.sites',
 'wagtail.users',
 'wagtail.snippets',
 'wagtail.documents',
 'wagtail.images',
 'wagtail.search',
 'wagtail.admin',
 'wagtail.core',
 'wagtail.contrib.modeladmin',
 'wagtailmenus',
 'wagtailmedia',
 'wagtailmarkdown',
 'wagtail_transfer',
 'wagtailsvg',
 'wagtail.contrib.settings',
 'django_comments_xtd',
 'django_comments',
 'modelcluster',
 'taggit',
 'allauth',
 'allauth.account',
 'allauth.socialaccount',
 'rest_framework',
 'sass_processor',
 'translation_manager',
 'django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.sites',
 'django_extensions']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'whitenoise.middleware.WhiteNoiseMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'iogt.middleware.LocaleMiddleware',
 'iogt.middleware.AdminLocaleMiddleware',
 'wagtail.contrib.redirects.middleware.RedirectMiddleware',
 'iogt_users.middlewares.RegistrationSurveyRedirectMiddleware',
 'external_links.middleware.RewriteExternalLinksMiddleware',
 'iogt.middleware.CacheControlMiddleware']



Traceback (most recent call last):
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/db/models/fields/__init__.py", line 1774, in get_prep_value
    return int(value)

The above exception (invalid literal for int() with base 10: '') was the direct cause of the following exception:
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/views/generic/base.py", line 70, in view
    return self.dispatch(request, *args, **kwargs)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/views/generic/base.py", line 98, in dispatch
    return handler(request, *args, **kwargs)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/views/generic/base.py", line 159, in get
    context = self.get_context_data(**kwargs)
  File "/data/idems/iogt/iogt/iogt/views.py", line 41, in get_context_data
    page = get_object_or_404(Page, pk=self.request.GET['page'])
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/shortcuts.py", line 76, in get_object_or_404
    return queryset.get(*args, **kwargs)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/db/models/query.py", line 418, in get
    clone = self._chain() if self.query.combinator else self.filter(*args, **kwargs)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/db/models/query.py", line 942, in filter
    return self._filter_or_exclude(False, *args, **kwargs)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/db/models/query.py", line 962, in _filter_or_exclude
    clone._filter_or_exclude_inplace(negate, *args, **kwargs)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/db/models/query.py", line 969, in _filter_or_exclude_inplace
    self._query.add_q(Q(*args, **kwargs))
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/db/models/sql/query.py", line 1360, in add_q
    clause, _ = self._add_q(q_object, self.used_aliases)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/db/models/sql/query.py", line 1379, in _add_q
    child_clause, needed_inner = self.build_filter(
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/db/models/sql/query.py", line 1321, in build_filter
    condition = self.build_lookup(lookups, col, value)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/db/models/sql/query.py", line 1167, in build_lookup
    lookup = lookup_class(lhs, rhs)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/db/models/lookups.py", line 24, in __init__
    self.rhs = self.get_prep_lookup()
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/db/models/lookups.py", line 76, in get_prep_lookup
    return self.lhs.output_field.get_prep_value(self.rhs)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/db/models/fields/__init__.py", line 1776, in get_prep_value
    raise e.__class__(

Exception Type: ValueError at /en/translation-not-found/
Exception Value: Field 'id' expected a number but got ''.
```

**Pull request merged**